ee/clickhouse/queries/clickhouse_retention.py

```diff
@@ -1,181 +1,78 @@
-    INITIAL_INTERVAL_SQL,
-    REFERENCE_EVENT_SQL,
-    REFERENCE_EVENT_UNIQUE_SQL,
+    RETENTION_BREAKDOWN_ACTOR_SQL,
     RETENTION_BREAKDOWN_SQL,
-    RETENTION_PEOPLE_SQL,
-    RETENTION_SQL,
 )
 from posthog.constants import (
     RETENTION_FIRST_TIME,
-    TREND_FILTER_TYPE_ACTIONS,
-    TREND_FILTER_TYPE_EVENTS,
-    TRENDS_LINEAR,
     RetentionQueryType,
 )
-from posthog.models.action import Action
-from posthog.models.entity import Entity
 from posthog.models.filters import RetentionFilter
-from posthog.models.person import Person
+from posthog.models.filters.retention_filter import RetentionPeopleRequest
 from posthog.models.team import Team
 from posthog.queries.retention import AppearanceRow, Retention
 
 CohortKey = NamedTuple("CohortKey", (("breakdown_values", Tuple[str]), ("period", int)))
 
 
 class ClickhouseRetention(Retention):
-    def _get_retention_by_cohort(self, filter: RetentionFilter, team: Team,) -> Dict[Tuple[int, int], Dict[str, Any]]:
-        period = filter.period
-        is_first_time_retention = filter.retention_type == RETENTION_FIRST_TIME
-        date_from = filter.date_from
-        trunc_func = get_trunc_func_ch(period)
-
-        returning_event_query, returning_event_params = RetentionEventsQuery(
-            filter=filter, team_id=team.pk, event_query_type=RetentionQueryType.RETURNING
-        ).get_query()
-        target_event_query, target_event_params = RetentionEventsQuery(
-            filter=filter,
-            team_id=team.pk,
-            event_query_type=RetentionQueryType.TARGET_FIRST_TIME
-            if is_first_time_retention
-            else RetentionQueryType.TARGET,
-        ).get_query()
-
-        all_params = {
-            "team_id": team.pk,
-            "start_date": date_from.strftime(
-                "%Y-%m-%d{}".format(" %H:%M:%S" if filter.period == "Hour" else " 00:00:00")
-            ),
-            **returning_event_params,
-            **target_event_params,
-            "period": period,
-        }
-
-        result = sync_execute(
-            RETENTION_SQL.format(
-                returning_event_query=returning_event_query,
-                trunc_func=trunc_func,
-                target_event_query=target_event_query,
-            ),
-            all_params,
-        )
-
-        initial_interval_result = sync_execute(
-            INITIAL_INTERVAL_SQL.format(reference_event_sql=target_event_query, trunc_func=trunc_func,), all_params,
-        )
-
-        result_dict = {}
-        for initial_res in initial_interval_result:
-            result_dict.update({(initial_res[0], 0): {"count": initial_res[1], "people": []}})
-
-        for res in result:
-            result_dict.update({(res[0], res[1]): {"count": res[2], "people": []}})
-
-        return result_dict
+    def run(self, filter: RetentionFilter, team: Team, *args, **kwargs) -> List[Dict[str, Any]]:
+        retention_by_breakdown = self._get_retention_by_breakdown_values(filter, team)
```

Comment:
This naming is confusing because it appears as if this function will only return breakdown based retention